### DISCUSSION TITLE
Again, it doesn't support the "@media" rule.

### DISCUSSION THREAD

**Discussion by Yari27** [cefsharp/CefSharp#4605] [General]
### Is there an existing issue for this?

- [X] I have searched both open/closed issues, no issue already exists.

### CefSharp Version

117.2.40

### Operating System

Windows 10

### Architecture

x64

### .Net Version

.Net 4.8

### Implementation

WPF

### Reproduction Steps

`
@media (prefers-color-scheme: dark) {
 body {
  color: #DEDEDE:
  background-color: #494949;
 }
}
`

### Expected behavior

Use dark body colors.

### Actual behavior

Don't change body colors.

### Regression?

Work in 115.3.110

### Does this problem also occur in the CEF Sample Application

Yes using WPF/OffScreen command line args

**Comment by amaitland** [comment]
The previous upstream issue was https://github.com/chromiumembedded/cef/issues/3449

> ### Does this problem also occur in the CEF Sample Application
> Yes using WPF/OffScreen command line args

Confirm that you've actually tested with the `CEF Sample Application`.  You can grab M117 builds from https://cef-builds.spotifycdn.com/index.html

Make sure to use the `WPF/OffScreen` command line args that were listed in the issue template.

Once you've confirmed the problem reproduces you'll need to create an issue on https://github.com/chromiumembedded/cef/issues referencing the previous issue.

**Comment by Yari27** [comment]
This display light theme google page.
`cefclient.exe --off-screen-rendering-enabled` and `cefclient.exe`
Only after refreshing it changes to dark theme.

This display dark theme page.
`cefclient.exe --url=https://dark-mode-baseline.glitch.me/ --off-screen-rendering-enabled`

**Reply by amaitland** [reply]
You've dropped some command line arguments 

```
 cefclient.exe --multi-threaded-message-loop --no-sandbox --off-screen-rendering-enabled --enable-gpu
```

The code for themes is at the `CEF/Chromium` layers, there's nothing in `CefSharp` that **directly** controls this behaviour.

You'll need to ask on https://www.magpcss.org/ceforum/index.php for further guidance.

**Reply by Yari27** [reply]
These arguments don't change anything :(

**Reply by amaitland** [reply]
You can **Git Bisect** the changes between [v115.3.110...v117.2.40](https://github.com/cefsharp/CefSharp/compare/v115.3.110...v117.2.40) to see exactly when the change in behaviour was introduced. Once you fix the exact commit you can compare the changes. 

For what it's worth CEF recently added a new command line arg to force light/dark mode, you can see how that behaves https://github.com/chromiumembedded/cef/issues/3534

> This display light theme google page. `cefclient.exe --off-screen-rendering-enabled` and `cefclient.exe` Only after refreshing it changes to dark theme.
> 
> This display dark theme page. `cefclient.exe --url=https://dark-mode-baseline.glitch.me/ --off-screen-rendering-enabled`

The difference is probably the url being passed to the browser upon creation vs loaded later.

**Reply by amaitland** [reply]
Testing with https://github.com/cefsharp/CefSharp.MinimalExample

Google shows in dark mode by default on my machine.

![image](https://github.com/cefsharp/CefSharp/assets/307872/598f2c9e-bb4c-441d-a178-644cffb48604)

Subsequently loading https://dark-mode-baseline.glitch.me/ also displays in dark mode.

![image](https://github.com/cefsharp/CefSharp/assets/307872/25ddb31f-6be4-4925-981a-40e36f0aeec8)

> Does the problem reproduce using the [MinimalExample](https://github.com/cefsharp/CefSharp.MinimalExample)? You can fork the MinimalExample and use this as a base for your example.

Did you test with the `MinimalExample` as per the bug report template? How does it behave?

For me everything appears to be working as expected.

**Comment by Yari27** [comment]
I rolled back to version 115.3.110 and checked each one in turn. The last one working is **116.0.15**, after this version it no longer works.

**Reply by amaitland** [reply]
> Testing with https://github.com/cefsharp/CefSharp.MinimalExample

Did you test with the `MinimalExample`?
